app/src/main/java/com/github/se/polyfit/ui/components/textField/MealInputTextField.kt

```diff
@@ -0,0 +1,101 @@
+package com.github.se.polyfit.ui.components.textField
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.SearchBar
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.polyfit.R
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.ui.theme.SecondaryGrey
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MealInputTextField(
+    meals: List<Meal>,
+    onCollapseSearchBar: () -> Unit = {},
+    onMealChange: (Meal) -> Unit = {}
+) {
+  val context = LocalContext.current
+  var showIngredientSearch by remember { mutableStateOf(false) }
+  var searchText by remember { mutableStateOf("") }
+  var searchResults by remember { mutableStateOf(meals) }
+
+  // Set the search bar to be at most 5 items tall, then scroll
+  val maxHeight = (74 + minOf(5, searchResults.size) * 41).dp
```

Comment:
Usually, it is best to find a more responsive way to do height, without hard calculations. I think this is ok for now, but after M2 maybe we can revisit this

Reply:
Ok. For now, it wasn't resizing well on its own and this was the best way